File: src/main/java/com/example/appforauslenderamt/entity/PlaceOfResidenceAbroad.java
package com.example.appforauslenderamt.entity;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

public class PlaceOfResidenceAbroad extends PlaceOfResidence{

    @JsonProperty("country")
    private final String country;

    @JsonCreator
    public PlaceOfResidenceAbroad(String postalCode, String place, String street, Integer houseNumber, String country) {
        super(postalCode, place, street, houseNumber);
        this.country = country;
    }

}

File: Frontend/src/app/Models/BackendModels/PlaceOfResidenceAbroad.ts
import { Expose } from "class-transformer"


export class PlaceOfResidenceAbroad {
    @Expose()
    country: string
    @Expose()
    postal_code: string
    @Expose()
    place: string
    @Expose()
    street: string
    @Expose()
    house_number: string


    constructor(
        country: string,
        postalCode: string,
        place: string,
        street: string,
        houseNumber: string
    ) {
        this.country = country
        this.postal_code = postalCode
        this.place = place
        this.street = street
        this.house_number = houseNumber
    }
}
